arelle/FunctionFn.py

```diff
@@ -307,8 +307,34 @@ def replace(xc, p, contextItem, args):
     except re.error as err:
         raise XPathContext.XPathException(p, 'err:FORX0002', _('fn:replace regular expression pattern error: {0}').format(err))
 
+
 def tokenize(xc, p, contextItem, args):
-    raise fnFunctionNotAvailable()
+    # See https://www.w3.org/TR/xpath-functions/#func-tokenize
+    if not 1 <= len(args) <= 3:
+        raise XPathContext.FunctionNumArgs()
+
+    if 1 == len(args):
+        input = normalize_space(xc, p, contextItem, args)
+        pattern = ' '
+    else:
+        input = stringArg(xc, args, 0, "xs:string?", emptyFallback="")  # empty string is default
+        pattern = stringArg(xc, args, 1, "xs:string", emptyFallback="")
+
+    if "" == input:
+        return []
+
+    flags = regexFlags(xc, p, args, 2)
+    flags |= re.V1 # Enable zero width matches
+
+    m_flags = "" if len(args) < 3 else args[2]
+    if matches(xc, p, contextItem, ["", pattern, m_flags]):
+        raise XPathContext.XPathException(p, 'err:FORX0003', _('fn:tokenize $pattern matches a zero-length string: {0}').format(pattern))
+
+    try:
+        return re.split(pattern, input, flags=flags)
```

Comment:
Hi Stuart,

Thank you for your contribution! It was great talking to you at XBRL Europe. I hope your journey home was smooth.

I believe we can handle capturing groups with the following code, but I'm also fine with accepting your current submission and addressing this in a follow-up:
```suggestion
        result = []
        lastEnd = 0
        for match in re.finditer(pattern, input, flags=flags):
            start, end = match.span()
            result.append(input[lastEnd:start])
            lastEnd = end
        result.append(input[lastEnd:])
        return result
```

Reply:
Hi Austin,

It was a great conversation! Yeah all well thanks and I hope your own travels go well this summer.

That's a great suggestion. I've just pushed a further commit incorporating it.